pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/AggregateNode.java

```diff
@@ -35,30 +35,72 @@ public class AggregateNode extends AbstractPlanNode {
   private List<RexExpression> _aggCalls;
   @ProtoProperties
   private List<RexExpression> _groupSet;
+  @ProtoProperties
+  private AggregationStage _aggregationStage;
+  @ProtoProperties
+  private boolean _treatIntermediateStageAsLeaf;
+
+  /**
+   * Enum to denote the aggregation stage being performed. Hints are used to populate these values
+   * LEAF - leaf aggregation level which performs the aggregations on the raw values directly
+   * INTERMEDIATE - intermediate aggregation level which merges results from lower aggregation levels
+   * FINAL - final aggregation level which extracts the final result
+   */
+  public enum AggregationStage {
+    LEAF,
+    INTERMEDIATE,
+    FINAL
+  }
```

Comment:
let's merge this ENUM with AggregateOperator.Mode
IMO, there's nothing we can't determine during plan time for runtime operator to perform.